code/lib/core-common/src/presets.ts

```diff
@@ -149,7 +150,13 @@ export const resolveAddonName = (
         ? {
             previewAnnotations: [
               previewFileAbsolute
-                ? { bare: previewFile, absolute: previewFileAbsolute }
+                ? {
+                    // TODO: Evaluate if searching for node_modules in a yarn pnp environment is correct
+                    bare: previewFile.includes('node_modules')
+                      ? stripAbsNodeModulesPath(previewFile)
```

Comment:
It's unfortunate that this is necessary, but it is caused by https://github.com/storybookjs/storybook/pull/21197#discussion_r1114545047 in all projects but also could happen if the user specifies addons in their main.js using absolute paths (for yarn pnp, for instance).

I would love to find a way to avoid so many absolute paths, but it seems like for now at least it's what we are stuck with.

Note that windows will likely still break if the user specifies absolute paths to non-node_modules files, but I don't think that should not be supported anyway (I can't think of a reason it would be needed, at least).

Reply:
In a pnp situation, the path won't include `node_modules`.